javascript/ql/src/semmle/javascript/frameworks/UrlRequests.qll

```diff
@@ -0,0 +1,127 @@
+/**
+ * Provides classes for modelling URL requests.
+ *
+ * Subclass `UrlRequest` to refine the behavior of the analysis on existing URL requests.
+ * Subclass `CustomUrlRequest` to introduce new kinds of URL requests.
+ */
+
+import javascript
+
+/**
+ * A call that performs a request to a URL.
+ */
+class CustomUrlRequest extends DataFlow::CallNode {
```

Comment:
This class conceptually overlaps with `NodeJSLib::ClientRequest` and its subclasses (recently introduced by @rdmarsh2). Could you look into combining the two?